plutus-core/plutus-core/src/PlutusCore/Constant/Typed.hs

```diff
@@ -533,18 +534,18 @@ class (uni ~ UniOf term, KnownTypeAst uni a) => KnownTypeIn uni term a where
     -- | Convert a Haskell value to the corresponding PLC term.
     -- The inverse of 'readKnown'.
     makeKnown
-        :: ( MonadEmitter m, MonadError (ErrorWithCause err cause) m, AsEvaluationFailure err
+        :: ( MonadError (ErrorWithCause err cause) m, AsEvaluationFailure err
            )
-        => Maybe cause -> a -> m term
+        => (Text -> m ()) -> Maybe cause -> a -> m term
```

Comment:
Kinda makes me wonder if instead of requiring `m` to implement `MonadError` and passing around that dictionary with `catchError` we could just pass another function explicitly.

Reply:
> "passing `catchError`" 

I mean that we don't need to pass `catchError` to that function at all, we don't do any error-catching in builtins from what I can remember (we do that outside of the builtins machinery in the CEK machine).

Or we could use `MonadThrow`.